Source/Core/UICommon/ResourcePack/ResourcePack.cpp

```diff
@@ -139,14 +141,44 @@ bool ResourcePack::Install(const std::string& path)
     return false;
   }
 
+  // TODO would strm_buffered interface give better perf?
   auto file = unzOpen(m_path.c_str());
+  if (file == nullptr)
+  {
+    m_valid = false;
+    m_error = "Failed to open resource pack";
+    return false;
+  }
   Common::ScopeGuard file_guard{[&] { unzClose(file); }};
 
-  for (const auto& texture : m_textures)
+  if (unzGoToFirstFile(file) != MZ_OK)
+    return false;
+  do
   {
-    bool provided_by_other_pack = false;
+    std::string texture_zip_path(UINT16_MAX + 1, '\0');
```

Comment:
Is `unzGetCurrentFileInfo64` just filling out the texture_zip_path data?  If so, could we move this out of the loop?

Reply:
i think since it's a std::string, some operations i'm doing on it could technically cause it to be reallocated (to a smaller size). so it would need to be `.resize`d each iteration. it may still be faster, i guess.